### ISSUE TITLE
Does it work for Meteor 1.2

### ISSUE THREAD

**Issue by ShayMatasaro** [kadirahq/react-komposer#33]
Hi,

I tried Komposer version 1.3.1 with Meteor 1.2 , but couldnt get it to work in a consistent manner.

A subscription would stop "on-its-own", 
The Tracker computations state had first-run = true on multiple runs

putting my function inside a simple Tracker.autorun worked and generated consistent results 

Thanks,
Shay

**Comment by arunoda** [comment]
Not tested with it. How do you  use it? With browserify? 
Can I have some sample app?

**Comment by ShayMatasaro** [comment]
Yes , I used browserify
Komposer = require('react-komposer');

the code was identical to the meteor sample from the docs :
function composer(props, onData) {
  if (Meteor.subscribe('posts').ready()) {
    const posts = Posts.find().fetch();
    onData(null, {posts});
  };
};
I would then notice that after the subscribe became ready , it would get removed, and any updates to the collection were not visible because the subscription was no longer active

I assume that the collection was un-subscribed due to  Autorun block closing , but couldn't see what caused it to close.

According to Auto-run docs if its the same subscription it should not get destroyed and recreated.

**Comment by clayne11** [comment]
I had a similar issue using Meteor 1.2.

When I had composeWithTracker nested under own HOC that used ReactMeteorData the subscriptions would sometimes get stopped. It would consistently happen when I opened a new browser tab going to the site.

**Comment by clayne11** [comment]
My issues seem to be resolved with some refactoring to my codebase. I don't think the issue was related to react-komposer.

**Comment by clayne11** [comment]
OK I'm actually still seeing this issue. Subscriptions will randomly stop when using composeWIthTracker.

**Comment by clayne11** [comment]
I've confirmed that using compose() with a Tracker.autorun inside of it works fine.

Using compseWithTracker, which looks like it stops the autorun block on each cycle and then restarts it is causing issues with my subscription getting stopped and started and constantly reloading data.

**Comment by clayne11** [comment]
More testing and I think I have come up with the real culprit. When I put Meteor.user() in a compseWithTracker composed on top of another HOC (as in the user gets fetched and passed into the next component to fetch data) I see a lot of issues my subscription constantly re-running. I'm not sure what would be causing Meteor.user() to change so frequently but it seems that it's constantly causing props changed to be fired.

**Comment by newsiberian** [comment]
Hello, I faced with this issue on with the latest Meteor (1.3-beta-12), and it happens all the time within one of the routes.
repo: [here](https://github.com/newsiberian/33-react-komposer-repro)
##### How to reproduce:
* This is reaction. You'll need to login as admin. Admin login/password will be visible in console after `meteor` (no need to do anything else)
* after login, go to `/shop` route, you should see one default product.
* step into it. Bug occurs [here](https://github.com/newsiberian/33-react-komposer-repro/blob/master/packages/reaction-react/imports/modules/products/containers/ProductDetailContainer.jsx#L97).

Within `composer` function you can check `Products` and `Media` collections by `ReactionCore.Collections.Products.find()` and `ReactionCore.Collections.Media.find()`, but after you step out this function, both collections become unreachable.

**Comment by newsiberian** [comment]
Hi, @arunoda, could you, please, take a look on this?

**Comment by arunoda** [comment]
@newsiberian Could you send me a small repo with the bug. It's hard for me to debug this big app :) 

@clayne11 can we have a sample repo. I'm keen to find this and fix it.

**Comment by newsiberian** [comment]
@arunoda, I'm sorry, but can't. I was tried to reproduce this on simple app before publishing this one, and on a simple app `komposer` worked as expected.

**Comment by clayne11** [comment]
I seem to have solved the issue. I never had any problem with the ReactMeteorData mixin so I looked at how that was constructed to get some insight.

They wrap their `Tracker.autorun` in a `Tracker.nonreactive` block. By doing this, I was able to fix all of the issues I was having with `composeWithTracker`.

The new `composeWithTracker` function looks like this:

```
export function composeWithTracker(reactiveFn, L, E, options) {
  const onPropsChange = (props, onData, context) => {
    let trackerCleanup;
    const handler = Tracker.nonreactive(() => {
      return Tracker.autorun(() => {
        trackerCleanup = reactiveFn(props, onData, context);
      });
    });

    return () => {
      if (typeof (trackerCleanup) === 'function') {
        trackerCleanup();
      }
      return handler.stop();
    };
  };

  return compose(onPropsChange, L, E, options);
}
```

I'm not 100% sure that the autorun is getting cleaned up properly though, however this has fixed all of my issues with data not being fetched properly and subscriptions getting stopped.

I put a print statement inside the inner `Tracker.autorun` to print the computation._id and I never saw the same _id getting printed after the component had been unmounted.

I'm going to submit a PR to with my fix and I'd like to hear your feedback on if you think I am cleaning up the autorun properly.
